src/commands/generate/fromTemplate.ts

```diff
@@ -0,0 +1,294 @@
+    return {
+      params: this.paramParser(params),
+      disableHooks: this.disableHooksParser(disableHooks),
+      mapBaseUrlToFolder: this.mapBaseURLParser(mapBaseUrl),
+    } as ParsedFlags;
+  }
+
+  private paramParser(inputs?: string[]) {
+    if (!inputs) { return {}; }
+    const params: Record<string, any> = {};
+    for (const input of inputs) {
+      if (!input.includes('=')) { throw new Error(`Invalid param ${input}. It must be in the format of --param name1=value1 name2=value2 `); }
+      const [paramName, paramValue] = input.split(/=(.+)/, 2);
+      params[String(paramName)] = paramValue;
+    }
+    return params;
+  }
+
+  private disableHooksParser(inputs?: string[]) {
+    if (!inputs) { return {}; }
+    const disableHooks: Record<string, any> = {};
+
+    for (const input of inputs) {
+      const [hookType, hookNames] = input.split(/=/);
+      if (!hookType) { throw new Error('Invalid --disable-hook flag. It must be in the format of: --disable-hook <hookType> or --disable-hook <hookType>=<hookName1>,<hookName2>,...'); }
+      if (hookNames) {
+        disableHooks[String(hookType)] = hookNames.split(',');
+      } else {
+        disableHooks[String(hookType)] = true;
+      }
+    }
+    return disableHooks;
+  }
+
+  private mapBaseURLParser(input?: string) {
+    if (!input) { return; }
+    const mapBaseURLToFolder: any = {};
+    const re = /(.*):(.*)/g; // NOSONAR
+    let mapping: any[] | null = [];
+    if ((mapping = re.exec(input)) === null || mapping.length !== 3) {
+      throw new Error('Invalid --map-base-url flag. A mapping <url>:<folder> with delimiter : expected.');
+    }
+
+    mapBaseURLToFolder.url = mapping[1].replace(/\/$/, '');
+    mapBaseURLToFolder.folder = path.resolve(mapping[2]);
+
+    const isURL = /^https?:/;
+    if (!isURL.test(mapBaseURLToFolder.url.toLowerCase())) {
+      throw new Error('Invalid --map-base-url flag. The mapping <url>:<folder> requires a valid http/https url and valid folder with delimiter `:`.');
+    }
+
+    return mapBaseURLToFolder;
+  }
+
+  private async generate(asyncapi: string | undefined, template: string, output: string, options: any, genOption: any) {
+    let specification: Specification;
+    try {
+      specification = await load(asyncapi);
+    } catch (err: any) {
+      return this.error(
+        new ValidationError({
+          type: 'invalid-file',
+          filepath: asyncapi,
+        }),
+        { exit: 1 },
+      );
+    }
+    const generator = new AsyncAPIGenerator(template, output || path.resolve(os.tmpdir(), 'asyncapi-generator'), options);
+
+    CliUx.ux.action.start('Generation in progress. Keep calm and wait a bit');
+    try {
+      await generator.generateFromString(specification.text(), genOption);
+      CliUx.ux.action.stop();
+    } catch (err: any) {
+      CliUx.ux.action.stop('done\n');
+      throw new GeneratorError(err);
```

Comment:
Created a new error for generator errors, and removed the error stack only printing the error message 

For Example: 
```
Generator Error: "docs" is in a git repository with unstaged changes. Please commit your changes before proceeding or add proper directory to .gitignore file. You can also use the --force-write flag to skip this rule (not recommended).
```